github.go

```diff
@@ -97,57 +97,79 @@ func NewGithubClient(s *Source) (*GithubClient, error) {
 }
 
 // ListOpenPullRequests gets the last commit on all open pull requests.
-func (m *GithubClient) ListOpenPullRequests() ([]*PullRequest, error) {
-	var query struct {
-		Repository struct {
-			PullRequests struct {
-				Edges []struct {
-					Node struct {
-						PullRequestObject
-						Commits struct {
-							Edges []struct {
-								Node struct {
-									Commit CommitObject
+func (m *GithubClient) ListOpenPullRequests() (response []*PullRequest, err error) {
+	var (
+		query struct {
+			Repository struct {
+				PullRequests struct {
+					Edges []struct {
+						Node struct {
+							PullRequestObject
+							Commits struct {
+								Edges []struct {
+									Node struct {
+										Commit CommitObject
+									}
+								}
+							} `graphql:"commits(last:$commitsLast)"`
+							Labels struct {
+								Edges []struct {
+									Node struct {
+										LabelObject
+									}
 								}
-							}
-						} `graphql:"commits(last:$commitsLast)"`
+							} `graphql:"labels(first:$labelsFirst)"`
+						}
 					}
-				}
-				PageInfo struct {
-					EndCursor   githubv4.String
-					HasNextPage bool
-				}
-			} `graphql:"pullRequests(first:$prFirst,states:$prStates,after:$prCursor)"`
-		} `graphql:"repository(owner:$repositoryOwner,name:$repositoryName)"`
-	}
+					PageInfo struct {
+						EndCursor   githubv4.String
+						HasNextPage bool
+					}
+				} `graphql:"pullRequests(first:$prFirst,states:$prStates,after:$prCursor)"`
+			} `graphql:"repository(owner:$repositoryOwner,name:$repositoryName)"`
+		}
 
-	vars := map[string]interface{}{
-		"repositoryOwner": githubv4.String(m.Owner),
-		"repositoryName":  githubv4.String(m.Repository),
-		"prFirst":         githubv4.Int(100),
-		"prStates":        []githubv4.PullRequestState{githubv4.PullRequestStateOpen},
-		"prCursor":        (*githubv4.String)(nil),
-		"commitsLast":     githubv4.Int(1),
-	}
+		queryVars = map[string]interface{}{
+			"repositoryOwner": githubv4.String(m.Owner),
+			"repositoryName":  githubv4.String(m.Repository),
+			"prFirst":         githubv4.Int(100),
+			"prStates":        []githubv4.PullRequestState{githubv4.PullRequestStateOpen},
+			"prCursor":        (*githubv4.String)(nil),
+			"commitsLast":     githubv4.Int(1),
+			"labelsFirst":     githubv4.Int(10),
+		}
+	)
 
-	var response []*PullRequest
 	for {
-		if err := m.V4.Query(context.TODO(), &query, vars); err != nil {
+		if err = m.V4.Query(context.TODO(), &query, queryVars); err != nil {
 			return nil, err
 		}
+
 		for _, p := range query.Repository.PullRequests.Edges {
-			for _, c := range p.Node.Commits.Edges {
```

Comment:
https://github.com/telia-oss/github-pr-resource/pull/132/files#diff-c411124e4d43a847f2f6360d74fba17eR138 we hard code to only get the last commit so I removed the loop. Still doesn't feel pretty so if you have any suggestions to improve this part they are more than welcome!

Reply:
Your solution should work, but ideally I'd like to avoid changing the way this loop works since it is tasked with the only thing where this resource really needs to be consistent 😅 

Since labels are set per PR and not commit, and should be the same for every `&PullRequest{}` (only the `Tip` would differ if for some reason we got multiple commits back), you can just create a list of labels prior to entering the loop over commit edges:

```go
		var response []*PullRequest
	for {
		if err := m.V4.Query(context.TODO(), &query, vars); err != nil {
			return nil, err
		}
		for _, p := range query.Repository.PullRequests.Edges {
			labels := make([]LabelObject, len(p.Node.Labels.Edges))

			for _, l := range p.Node.Labels.Edges {
				labels = append(labels, l.Node.LabelObject)
			}

			for _, c := range p.Node.Commits.Edges {
				response = append(response, &PullRequest{
					PullRequestObject: p.Node.PullRequestObject,
					Tip:               c.Node.Commit,
					Labels:            labels,
				})
			}
		}
		if !query.Repository.PullRequests.PageInfo.HasNextPage {
			break
		}
		vars["prCursor"] = query.Repository.PullRequests.PageInfo.EndCursor
	}
```

Should work no?

Reply:
Yep definitely :) Thank you!